src/cbmc/cbmc_parse_options.cpp

```diff
@@ -225,10 +225,12 @@ void cbmc_parse_optionst::get_command_line_options(optionst &options)
   if(cmdline.isset("no-simplify"))
     options.set_option("simplify", false);
 
-  if(cmdline.isset("stop-on-fail") ||
-     cmdline.isset("dimacs") ||
-     cmdline.isset("outfile"))
+  if(
+    cmdline.isset("stop-on-fail") || cmdline.isset("dimacs") ||
+    (cmdline.isset("outfile") && !cmdline.isset("incremental-smt2-solver")))
+  {
     options.set_option("stop-on-fail", true);
+  }
```

Comment:
I'm either failing to parse the commit message or don't understand why the incremental SMT2 solver now interacts with "outfile" in this way. Is it the case that "outfile" has a very different semantics with the incremental SMT2 solver?! Does not seem like a good idea.

Reply:
To your "somewhat related question:" no actual solving takes place with other back-ends with `--outfile`, and also: what would be the semantics of having only a single output file but multiple formulae having to be written?

So if the incremental SMT2 back-end has substantially different behaviour (solving actually takes place, multiple formulae are written) then we really shouldn't be using the same command-line option name. This confusing user-experience is a blocker.

Reply:
>Otherwise only the first round of solving is written to file, because we stopped on failure.

But if you specified --stop-on-fail then this is exactly what you asked for. So, I don't se the point of having special behaviour here, @thomasspriggs.

Reply:
But this is for when we aren't explicitly specifying `--stop-on-fail`. The argument parsing is currently adding "stop-on-fail" implicitly when `--outfile` is specified. Do we need to add an explicit `--continue-on-fail` to override the implicit option?

Reply:
I see. Then I'd call the current behaviour somewhat confusing... not sure how to fix that without having different behaviour for incremental vs non-incremental, or having multiple options.